### PR TITLE
Steed - add type definition

### PR DESCRIPTION
**Pull request by Paul-Isache**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
-  [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<enter here>>
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: steed/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    series<T, E>(tasks: Array<SteedFunction<T, E>>, callback?: SteedResultArrayCallback<T, E>): void;
+    series<T, E>(tasks: Dictionary<SteedFunction<T, E>>, callback?: SteedResultObjectCallback<T, E>): void;
+    waterfall<T, E>(tasks: Function[], callback?: SteedResultCallback<T, E>): void;
+    each<T, E>(arr: T[] | Dictionary<T>, iterator: SteedIterator<T, E>, callback?: ErrorCallback<E>): void;
+    eachSeries: typeof steed.each;
+    map<T, R, E>(arr: T[] | Dictionary<T>, iterator: SteedResultIterator<T, R, E>, callback?: SteedResultArrayCallback<R, E>): void;
+    mapSeries: typeof steed.map;
+    queue<T, E>(worker: SteedWorker<T, E>, concurrency?: number): SteedQueue<T>;
+    queue<T, R, E>(worker: SteedResultIterator<T, R, E>, concurrency?: number): SteedQueue<T>;
+}
```
Comment (DanielRosenwasser) [inline steed/index.d.ts]: These interfaces aren't publicly available, so can you wrap them in a `declare namespace steed { /*...*/ }`? Then make the below

```ts
declare const steed: steed.Steed;
```

File: steed/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+declare const steed: Steed;
+
+declare module "steed" {
```
Comment (DanielRosenwasser) [inline steed/index.d.ts]: Unwrap this to the top level as `export = steed`

File: steed/steed-tests.ts [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+import process = require("process");
+
+declare var path: {
+    exists: (path: string, callback?: (err: Error, exists: boolean) => any) => void;
+};
+
+function funcStringCbErrBoolean(v:string, cb:(err:Error,res:boolean) => void) {}
+function callback() { }
+
+steed.map(['file1', 'file2', 'file3'], fs.stat, function (err:Error, results:Array<fs.Stats>) { });
```
Comment (DanielRosenwasser) [inline steed/steed-tests.ts]: The package doesn't seem to be exported as a global - please import it.

File: steed/steed-tests.ts [hunk 4]
```diff
@@ -0,0 +1,353 @@
+/// <reference types="node" />
+
+import fs = require("fs");
```
Comment (DanielRosenwasser) [inline steed/steed-tests.ts]: Please don't bring in `fs` and `process` if the original file doesn't use them.

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by DanielRosenwasser** [comment]
Just as a heads up, the email on these commits isn't registered with GitHub - if you need to rebase and fixup the author, or register the email you used with GitHub, you migth want to do that. If you don't care much, i can just merge it in.

**Comment by Paul-Isache** [comment]
You can merge it. There is an issue with my account on my laptop and on the future I will try and fix it, but for now si good

**Review by DanielRosenwasser — APPROVED** [review]

**Comment by DanielRosenwasser** [comment]
Thanks!

**Pull request merged**